### PR TITLE
Added APIService

### PR DESCRIPTION
**Pull request by epicadk**
### Description

Added APIService

### Changes Done

No

### Acknowledgements

Please ensure the following and put a check accordingly

- [x] My code doesn't break/affect any unrelated part of the codebase

- [x] Squashed multiple commits (_if irrelevant_) into one commit

### CODE REVIEW

File: app/src/main/java/com/tip/lunchbox/data/service/APIService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import io.reactivex.rxjava3.core.Observable;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+import retrofit2.http.QueryMap;
+
+public interface APIService {
+
+    @GET("search")
+    Observable<SearchResponse> getSearchResponse(@QueryMap Map<String, String> searchParameters);
```
Comment (devansh-299) [inline app/src/main/java/com/tip/lunchbox/data/service/APIService.java:19]: @epicadk can you please explain the logic of passing a query map? From the documentation, I can see that search can be used be in multiple ways, should separate methods be defined for each?

File: app/src/main/java/com/tip/lunchbox/data/service/APIService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @GET("dailymenu")
+    Observable<DailyMenuResponse> getDailyMenuResponse(@Query("res_id") int restaurantID);
+
+    @GET("reviews")
+    Observable<List<Reviews>> getReviewsResponse(@Query("res_id") int restaurantID,
+                                                 @Query("start") Integer start,
```
Comment (devansh-299) [inline app/src/main/java/com/tip/lunchbox/data/service/APIService.java]: I think we can avoid these `start` and `count` queries for now as the public key only provides 5 top reviews only

File: app/src/main/java/com/tip/lunchbox/data/service/APIService.java (hunk 1 above)
Comment (epicadk) [reply app/src/main/java/com/tip/lunchbox/data/service/APIService.java:19]: I thought we could do that in the repository class if required

Comment (epicadk) [reply app/src/main/java/com/tip/lunchbox/data/service/APIService.java:19]: Should that be in the repository or the api interface itself?

Comment (devansh-299) [reply app/src/main/java/com/tip/lunchbox/data/service/APIService.java:19]: We can create a separate interface with all the different searching use cases inside it as different methods? What do you think?

Comment (epicadk) [reply app/src/main/java/com/tip/lunchbox/data/service/APIService.java:19]: Yes but all search cases may not be used I suggest we leave it at this and if we do use a few of them more than once we create a separate Interface

**Pull request merged**